### PR TITLE
C#: Only consider latest version of dotnet framework flavors

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 1]
```diff
@@ -237,14 +237,32 @@ private void AddNetFrameworkDlls(ISet<string> dllPaths, ISet<string> frameworkLo
...
+                .FirstOrDefault(pair => pair.Path is not null);
 
             if (frameworkPath.Path is not null)
             {
-                dllPaths.Add(frameworkPath.Path);
-                frameworkLocations.Add(frameworkPath.Path);
-                progressMonitor.LogInfo($"Found .NET Core/Framework DLLs in NuGet packages at {frameworkPath.Path}. Not adding installation directory.");
+                var versionFolders = new DirectoryInfo(frameworkPath.Path)
+                    .EnumerateDirectories("*", new EnumerationOptions { MatchCasing = MatchCasing.CaseInsensitive, RecurseSubdirectories = false })
+                    .OrderByDescending(d => d.Name) // TODO: Improve sorting to handle pre-release versions.
```
Comment (michaelnebel) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:247]: Small comment: Sorting can be avoided to find the "correct" framework. This is not case, if it is important that the progressmonitor prints the version folder names in sorted order (I suppose this this the case).

**Review by michaelnebel — APPROVED** [review]
One more question: Do we need a similar change for adding the ASP.NET core framework dlls?

**Comment by tamasvajk** [comment]
> One more question: Do we need a similar change for adding the ASP.NET core framework dlls?

Yes, that's a good point. It's also needed for windows desktop dlls too. Let's do these in a followup PR.

**Comment by tamasvajk** [comment]
> > One more question: Do we need a similar change for adding the ASP.NET core framework dlls?
> 
> Yes, that's a good point. It's also needed for windows desktop dlls too. Let's do these in a followup PR.

In the end I added this change to the PR.

**Review by michaelnebel — APPROVED** [review]
Great!

**Pull request merged**